Sources/Channel.swift

```diff
@@ -27,18 +27,18 @@ import Swift
 public class Channel {
     
     /// The topic of the Channel. e.g. "rooms:friends"
-    let topic: String
+    public let topic: String
     
     /// The params sent when joining the channel
     var params: Payload
 
     /// The Socket that the channel belongs to
-    weak var socket: Socket?
+    public weak var socket: Socket?
```

Comment:
I don't like exposing the socket outside of the channel, especially since it is a `weak var` now and not an immutable `let` Also, you should already have access to the `socket` since it is required to create the channel in the first place

Reply:
I have a piece of code that I'm not sure how to rebuild otherwise:

```            guard !type(of: self).connectors.contains(where: { (connector) -> Bool in
                connector.channel?.topic == topic
            }) else {
                return
            }```

I built it before reconnection logic really worked though.

Could it be `private(set)` so it would be effectively read-only?

Reply:
It looks like you're accessing only the channel's `topic` in that piece of code, which I am fine with exposing. Am I reading that correctly?

Reply:
That's right. An alternative way to access it would be great

Reply:
I'm fine exposing `topic` as a `public let topic`, however you are also exposing `socket` as `public socket`, which I don't think should be allowed and it looks like from your code snippet above, you're not making use of `socket` anyways

Reply:
so in `Channel.swift` we would add something like:

```let topic: String```

which gets set in the constructor?

Do you want me to create a new branch and pull request?

Reply:
You've already made the change `public let topic: String` in this PR [at line 30](https://github.com/davidstump/SwiftPhoenixClient/pull/115/files#diff-0f6463bf4fb25f9638a8ad7fca2193ffR29). which I think is 👍 which makes `channel.topic` accessible

I would just like to see the `public` modifier removed from the socket declaration [at line 36](https://github.com/davidstump/SwiftPhoenixClient/pull/115/files#diff-0f6463bf4fb25f9638a8ad7fca2193ffR36) so that `channel.socket` is not accessible